driver-core/src/main/com/mongodb/internal/operation/ListSearchIndexesOperation.java

```diff
@@ -0,0 +1,132 @@
+import com.mongodb.internal.async.SingleResultCallback;
+import com.mongodb.internal.binding.AsyncReadBinding;
+import com.mongodb.internal.binding.ReadBinding;
+import com.mongodb.lang.NonNull;
+import com.mongodb.lang.Nullable;
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+import org.bson.BsonValue;
+import org.bson.codecs.Decoder;
+
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+
+import static com.mongodb.assertions.Assertions.assertNotNull;
+import static com.mongodb.internal.operation.CommandOperationHelper.isNamespaceError;
+import static com.mongodb.internal.operation.CommandOperationHelper.rethrowIfNotNamespaceError;
+import static com.mongodb.internal.operation.OperationHelper.createEmptyAsyncBatchCursor;
+import static com.mongodb.internal.operation.OperationHelper.createEmptyBatchCursor;
+
+/**
+ * An operation that lists Alas Search indexes with the help of {@value #STAGE_LIST_SEARCH_INDEXES} pipeline stage.
+ *
+ * <p>This class is not part of the public API and may be removed or changed at any time</p>
+ */
+public class ListSearchIndexesOperation<T>
+        implements AsyncExplainableReadOperation<AsyncBatchCursor<T>>, ExplainableReadOperation<BatchCursor<T>> {
+    private static final String STAGE_LIST_SEARCH_INDEXES = "$listSearchIndexes";
+    private final MongoNamespace namespace;
+    private final Decoder<T> decoder;
+    private final Boolean allowDiskUse;
+    private final Integer batchSize;
+    private final Collation collation;
+    private final BsonValue comment;
+    private final long maxTimeMS;
+    private final String indexName;
+
+    public ListSearchIndexesOperation(final MongoNamespace namespace,
+                                      final Decoder<T> decoder,
+                                      final long maxTimeMS,
+                                      @Nullable final String indexName,
+                                      @Nullable final Integer batchSize,
+                                      @Nullable final Collation collation,
+                                      @Nullable final BsonValue comment,
+                                      @Nullable final Boolean allowDiskUse) {
+        this.namespace = namespace;
+        this.decoder = decoder;
+        this.allowDiskUse = allowDiskUse;
+        this.batchSize = batchSize;
+        this.collation = collation;
+        this.maxTimeMS = maxTimeMS;
+        this.comment = comment;
+        this.indexName = indexName;
+    }
+
+    @Override
+    public BatchCursor<T> execute(final ReadBinding binding) {
+        try {
+            return asAggregateOperation().execute(binding);
+        } catch (MongoCommandException exception) {
+            int cursorBatchSize = batchSize == null ? 0 : batchSize;
+            return assertNotNull(rethrowIfNotNamespaceError(exception, createEmptyBatchCursor(namespace, decoder,
+                    exception.getServerAddress(), cursorBatchSize)));
+        }
+    }
```

Comment:
Similarly to `DropSearchIndexOperation`, using `rethrowIfNotNamespaceError` as a synchronous counterpart to `isNamespaceError` only harms readability: these methods are not alike in both their naming and behavior.

The code

```java
if (!isNamespaceError(exception)) {
    throw exception;
} else {
    return createEmptyBatchCursor(namespace, decoder, exception.getServerAddress(), cursorBatchSize);
}
```

is clearer than

```java
return assertNotNull(rethrowIfNotNamespaceError(exception, createEmptyBatchCursor(namespace, decoder,
        exception.getServerAddress(), cursorBatchSize)));
```

does not create an empty cursor when it is not needed, and, importantly, makes the synchronous and asynchronous code more similar to each other.